src/clients/oauth.rs

```diff
@@ -38,16 +38,25 @@ pub trait OAuthClient: BaseClient {
     /// token will be saved internally.
     async fn refresh_token(&mut self, refresh_token: &str) -> ClientResult<()>;
 
-    /// Tries to read the cache file's token, which may not exist.
+    /// Tries to read the cache file's token.
+    ///
+    /// This will return an error if the token couldn't be read (e.g. it's not
+    /// available or the JSON is malformed). It may return `Ok(None)` if:
+    ///
+    /// * The read token is expired
+    /// * The cached token is disabled in the config
```

Comment:
It seems you are missing one condition that
```rust
/// * Its scopes don't match with the current client
```

Reply:
Whoops, I messed up the docs. Should be fine now